src/plugins/core/image.ts

```diff
@@ -0,0 +1,119 @@
+}
+
+export class ImagePlugin extends CorePlugin<ImageState> implements ImageState {
+  static getters = ["getImageIds", "getImage", "getPath"] as const;
+
+  readonly images: Record<UID, Image | undefined> = {};
+  readonly nextId = 1;
+
+  // ---------------------------------------------------------------------------
+  // Command Handling
+  // ---------------------------------------------------------------------------
+
+  handle(cmd: CoreCommand) {
+    switch (cmd.type) {
+      case "CREATE_IMAGE_OVER":
+        this.addFigure(cmd.figureId, cmd.sheetId, cmd.position, cmd.size);
+        this.history.update("images", cmd.figureId, image(cmd.sheetId, cmd.definition.path));
+        break;
+      case "DUPLICATE_SHEET": {
+        const sheetFiguresFrom = this.getters.getFigures(cmd.sheetId);
+        for (const fig of sheetFiguresFrom) {
+          if (fig.tag === "image") {
+            const id = this.nextId.toString();
+            this.history.update("nextId", this.nextId + 1);
+            const image = this.images[fig.id];
+            if (image) {
+              const size = { width: fig.width, height: fig.height };
+              this.dispatch("CREATE_IMAGE_OVER", {
+                figureId: id,
+                position: { x: fig.x, y: fig.y },
+                size,
+                definition: {
+                  sheetId: cmd.sheetId,
+                  path: image.path,
+                },
+                sheetId: cmd.sheetIdTo,
+              });
+            }
+          }
+        }
+        break;
+      }
+      case "DELETE_FIGURE":
+        this.history.update("images", cmd.id, undefined);
+        break;
+      case "DELETE_SHEET":
+        for (const id of this.getImageIds(cmd.sheetId)) {
+          this.history.update("images", id, undefined);
+        }
+        break;
+    }
+  }
+
+  // ---------------------------------------------------------------------------
+  // Getters
+  // ---------------------------------------------------------------------------
+
+  getImageIds(sheetId: UID): UID[] {
+    return Object.entries(this.images)
+      .filter(([, image]) => image?.sheetId === sheetId)
+      .map(([id]) => id);
+  }
+
+  getImage(figureId: UID): Image {
+    const image = this.images[figureId];
+    if (!image) {
+      throw new Error(`There is no image with the given figureId: ${figureId}`);
+    }
+    return image;
+  }
+
+  getPath(figureId: UID): string {
+    return this.getImage(figureId)?.path;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Private
+  // ---------------------------------------------------------------------------
+
+  private addFigure(
+    id: UID,
+    sheetId: UID,
+    position: { x: Pixel; y: Pixel } = { x: 0, y: 0 },
+    size: { width: Pixel; height: Pixel } = {
+      width: DEFAULT_FIGURE_WIDTH,
+      height: DEFAULT_FIGURE_HEIGHT,
+    }
+  ) {
+    if (this.getters.getFigure(sheetId, id)) {
+      return;
+    }
```

Comment:
Seems dangerous, if a such figure exists it's because there is an issue somewhere. `allowDispatch` can help you